api/src/main/java/org/apache/iceberg/view/ViewVersion.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.view;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A version of the view at a point in time.
+ * <p>
+ * A version consists of a view metadata file.
+ * <p>
+ * Versions are created by view operations, like Create and Replace.
+ */
+public interface ViewVersion {
```

Comment:
To me, `version` only carries the meaning of some version id or number. Just brainstorming some ideas: `ViewPoint` or `PointOfView`?

Reply:
View Version does contain a version id which identifies the actual version of the view, but I think I get what you mean; the name "version" should be tied to just the numeric id. Some other ideas in my mind is `ViewInstance` or `ViewSnapshot` but maybe snapshot will get confusing, although that is what this really is at least in my mind.

Reply:
Yeah, version typically means just the numeric id to me. I also thought about `ViewSnapshot`. I think it can get confused with table snapshot. Or if we use git analogy, maybe `ViewCommit`?

Reply:
It is not snapshot. We deliberately chose `version` to differentiate from `snapshot`.
As for `ViewCommit`, unfortunately that is another over-used word :(

Reply:
I think `ViewVersion` is fine. We could also go with just `Version`, since this is in the `view` package.

Reply:
What about `Definition`? We can use a version ID for the definition, but call it a "definition".

Reply:
+1 Ryan's latest suggestion of `ViewDefinition`. It is the most intuitive name so far.

Reply:
TODO: sounds good, checking the code to be sure.

Reply:
Unfortunately, "view definition" has different meaning in the spec:

```
##### Original View Definition in SQL

This type of representation stores the original view definition in SQL and its SQL dialect.
```